### PR TITLE
add namespaceOverride support for privateRegistry

### LINKED ISSUE #940: Add namespaceOverride param in registry configuration
### Your current KubeKey version

latest

### Describe this feature

image
When I set privateRegistry with myself, but the default namespace is kubesphereio which does not exist.

Is it possible  adding a new param "namespaceOverride" to support namespace self defined.

### Describe the solution you'd like

add namespaceOverride param in registry configuration.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by life-**
Fix #940

### CODE REVIEW

**Comment by 24sama** [comment]
Thanks for your contribution!

**Comment by 24sama** [comment]
Could you please update this example file? https://github.com/kubesphere/kubekey/blob/master/docs/config-example.md

**Comment by life-** [comment]
> Could you please update this example file? https://github.com/kubesphere/kubekey/blob/master/docs/config-example.md

ok

File: apis/kubekey/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -177,6 +177,7 @@ type RegistryConfig struct {
 	RegistryMirrors    []string `yaml:"registryMirrors" json:"registryMirrors,omitempty"`
 	InsecureRegistries []string `yaml:"insecureRegistries" json:"insecureRegistries,omitempty"`
 	PrivateRegistry    string   `yaml:"privateRegistry" json:"privateRegistry,omitempty"`
+	NamespaceOverride  string   `yaml:"namespaceOverride" json:"namespaceOverride,omitempty"`
```
Comment (24sama) [inline apis/kubekey/v1alpha1/cluster_types.go:180]: The `v1alpha1` version of the struct does not require this field. You can remove it.

File: config/crd/bases/kubekey.kubesphere.io_clusters.yaml [hunk 2]
```diff
@@ -255,6 +255,8 @@ spec:
                     type: array
                   privateRegistry:
                     type: string
+                  namespaceOverride:
```
Comment (24sama) [inline config/crd/bases/kubekey.kubesphere.io_clusters.yaml:258]: This field also can be removed. The reason is the same as above.

File: manifests/deploy.yaml [hunk 3]
```diff
@@ -184,6 +184,8 @@ spec:
                   type: array
                 privateRegistry:
                   type: string
+                namespaceOverride:
```
Comment (24sama) [inline manifests/deploy.yaml:187]: This field also can be removed. The reason is the same as above. And I will update this deploy file later.

**Review by 24sama — CHANGES_REQUESTED** [review]

**Comment by 24sama** [comment]
> > Could you please update this example file? https://github.com/kubesphere/kubekey/blob/master/docs/config-example.md
> 
> ok

Thank you for the revision. And I add some review comments. Please take a look.

File: apis/kubekey/v1alpha1/cluster_types.go (hunk 1 above)
Comment (life-) [reply apis/kubekey/v1alpha1/cluster_types.go]: done

File: config/crd/bases/kubekey.kubesphere.io_clusters.yaml (hunk 2 above)
Comment (life-) [reply config/crd/bases/kubekey.kubesphere.io_clusters.yaml]: done

File: manifests/deploy.yaml (hunk 3 above)
Comment (life-) [reply manifests/deploy.yaml]: done

**Comment by life-** [comment]
@24sama Thanks for your code review, I have already done change according your review comments. Pls have a look.

**Comment by 24sama** [comment]
Hi @life- 
And one last thing, could you please squash your commits as one commit? That will make our commits look clearer.

**Comment by life-** [comment]
ok, I havae squashed my commits as one commit. Thanks!

**Comment by 24sama** [comment]
Thanks @life-  
/lgtm

**Pull request merged**